libraries/aws_cloudwatch_log_metric_filter.rb

```diff
@@ -0,0 +1,152 @@
+require 'aws_conn'
+
+class AwsCloudwatchLogMetricFilter < Inspec.resource(1)
+  name 'aws_cloudwatch_log_metric_filter'
+  desc 'Verifies individual Cloudwatch Log Metric Filters'
+  example <<-EOX
+  # Look for a LMF by its filter name and log group name.  This combination
+  # will always either find at most one LMF - no duplicates.
+  describe aws_cloudwatch_log_metric_filter(
+    filter_name: 'my-filter',
+    log_group_name: 'my-log-group'
+  ) do
+    it { should exist }
+  end
+
+  # Search for an LMF by pattern and log group.
+  # This could result in an error if the results are not unique.
+  describe aws_cloudwatch_log_metric_filter(
+    log_group_name:  'my-log-group',
+    pattern: 'my-filter'
+  ) do
+    it { should exist }
+  end
+EOX
+
+  RESOURCE_PARAMS = [
+    :filter_name,
+    :log_group_name,
+    :pattern,
+  ].freeze
+
+  attr_reader :found, :filter_name, :log_group_name, :pattern, :metric_name, :metric_namespace
+
+  def initialize(resource_params)
+    resource_params = validate_resource_params(resource_params)
+    results = run_lmf_search(resource_params)
+    if results.count > 1
```

Comment:
Do we need a case for `results.count == 0`?

Reply:
Maybe break all of these out into a `validate_search_results`?

Reply:
I've consolidated all of this into a `fetch` method, more like what aws_sns_topic does.